src/shared/condition.c

```diff
@@ -262,6 +263,44 @@ static int condition_test_kernel_version(Condition *c, char **env) {
         return true;
 }
 
+static int condition_test_osrelease(Condition *c, char **env) {
+        _cleanup_free_ char *key = NULL, *actual_value = NULL;
+        const char *parameter = c->parameter, *requested_value;
+        OrderOperator order;
+        int r;
+
+        assert(c);
+        assert(c->parameter);
+        assert(c->type == CONDITION_OS_RELEASE);
+
+        r = extract_first_word(&parameter, &key, "!<=>", 0);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to parse parameter: %m");
+        if (r < 1)
+                return log_debug_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "Failed to parse parameter, key/value format expected: %m");
+
+        /* parse_order() needs the string to start with the comparators, so reset the variable */
+        parameter = c->parameter + strlen(key);
```

Comment:
hmm, i think it would be much nicer to have another flag for extract_first_word(), maybe called EXTRACT_LEAVE_SEPARATOR or so, that skips over the word but not the follwing separator. You could then use this here and all would be good.

The thing is that in particular when quoting/escaping is on the strlen(key) thing will not actually match the length of the key really... Now, even if we decide not to support quoting/escaping here I figure we should still abstract that way